Shop_Unity_base/Assets/Scripts/Shop/Model/Components/IComponentOwner.cs

```diff
@@ -0,0 +1,19 @@
+﻿
+namespace Model
+{
+    using System;
+    using System.Collections.Generic;
+
+    public interface IComponentOwner<U>
```

Comment:
Perhaps constrain it more to tell the programmer inspecting this source code, how the interface is intrinsically meant to be used. Though, yeah obviously doesn't work so well with inheritance, but eh.
```suggestion
    public interface IComponentOwner<U> : where U : IComponentOwner<U>
```

Reply:
I don't see inheritance being too important here though, so I can do that. Derived classes will usually not overwrite AddComponent and it's siblings, like in Unity GameObject.